File: project_euler/Problem029/Problem029.go
package project_euler

import (
    "fmt"
    "math"
    "sort"
)

func Problem029() {
    start := 2
    end := 100
    terms := make([]float64, 0)
    for i := start; i <= end; i++ {
        for j := start; j <= end; j++ {
            terms = append(terms, float64(i) * math.Log(float64(j)))
        }
    }

    sort.Float64s(terms)

    result := len(terms)
    for i := result - 1; i >= 1; i-- {
        if math.Abs(terms[i] - terms[i - 1]) < 0.00000001 {
            result--
        }
    }
    fmt.Println(result)
}

File: project_euler/Problem029/Problem029.js
function Problem029() {
    let start = 2
    let end = 100
    let terms = []
    for (let i = start; i <= end; ++i) {
        for (let j = start; j <= end; ++j) {
            terms.push(i * Math.log(j))
        }
    }

    terms.sort()

    let result = terms.length
    for (let i = result - 1; i >= 1; --i) {
        if (Math.abs(terms[i] - terms[i - 1]) < 0.00000001) {
            --result
        }
    }
    console.log(result)
}

module.exports = {
    Problem029 : Problem029
};
